src/main/java/science/icebreaker/media/MediaService.java

```diff
@@ -0,0 +1,95 @@
+package science.icebreaker.media;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.nio.file.Files;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import org.springframework.web.multipart.MultipartFile;
+
+import science.icebreaker.exception.ErrorCodeEnum;
+import science.icebreaker.exception.StorageException;
+
+@Service
+public class MediaService {
+    private final String fileDirectory;
+    private final List<String> allowedTypes;
+    private MediaRepository repository;
+
+    public MediaService(
+        @Value("${icebreaker.files.dir}") String fileDir,
+        @Value("#{'${icebreaker.files.allowedTypes}'.split(',')}") List<String> allowedTypes,
+        MediaRepository repository
+    ) {
+        this.fileDirectory = fileDir;
+        this.allowedTypes = allowedTypes;
+        this.repository = repository;
+    }
+
+    /**
+     * Returns the directory of a given media by its identifier
+     * @param fileName the filename/identifier of the media
+     * @return The Path to the media file
+     */
+    public Path getMediaLocation(String fileName) {
+        return Paths
+            .get(fileDirectory
+                + File.separator
+                + StringUtils.cleanPath(fileName)
+            );
+    }
+
+    /**
+     * Stores a media file on disk
```

Comment:
What is the advantage of storing a Media File on the disk rather than a decoded version directly in the database?

Reply:
Less load of the backend itself. Depending on your filesystem, you can get some nice features such as deduplication.